service/history/ndc/history_replicator.go

```diff
@@ -660,11 +661,25 @@ func (r *HistoryReplicatorImpl) applyNonStartEventsToNonCurrentBranchWithoutCont
 		return err
 	}
 
-	transactionID, err := r.shard.GenerateTaskID()
+	var workflowEvents []*persistence.WorkflowEvents
+	transactionIDs, err := r.shard.GenerateTaskIDs(len(task.getHistories()))
 	if err != nil {
 		return err
 	}
-
+	var prevTxnId int64 = 0
+	for i := 0; i < len(task.getHistories()); i++ {
+		wfEvents := &persistence.WorkflowEvents{
+			NamespaceID: task.getNamespaceID().String(),
+			WorkflowID:  task.getExecution().GetWorkflowId(),
+			RunID:       task.getExecution().GetRunId(),
+			BranchToken: versionHistory.GetBranchToken(),
```

Comment:
I assume the batch of history will have the same branch token in replication. But not 100% sure.